### PR TITLE
build: add new build workflow

### PR DESCRIPTION
**Pull request by mehmetihsansevinc**
## What this PR changes/adds

_Added nightly.yaml workflow._

## Why it does that

_Builds and publishes at nights._

### CODE REVIEW

File: .github/workflows/verify.yaml [hunk 1]
```diff
@@ -31,6 +31,7 @@ jobs:
         run: ./gradlew checkstyleMain checkstyleTest
 
   Dependency-Check:
+    if: ${{ inputs.skip-dependency-check != 'true' }}
```
Comment (ndr-brt) [inline .github/workflows/verify.yaml:34]: no need for this, dependency-check won't run on scheduled workflows anyway (ref. https://github.com/eclipse-edc/.github/blob/dfb2f1c61348c41d35003b7ab04fc1dd33655077/.github/workflows/dependency-check.yml#L28)

**Review by ndr-brt — CHANGES_REQUESTED** [review]
old comment still not resolved

**Comment by mehmetihsansevinc** [comment]
> old comment still not resolved

skip dependency check line removed

File: .github/workflows/verify.yaml [hunk 2]
```diff
@@ -1,6 +1,12 @@
 name: Run Tests
 
 on:
+  workflow_call:
+    inputs:
+      skip-dependency-check:
+        description: 'Skip the Dependency-Check job'
+        required: false
+        default: 'false'
```
Comment (ndr-brt) [inline .github/workflows/verify.yaml:9]: also this is not needed

**Review by ndr-brt — CHANGES_REQUESTED** [review]
please re-request the review after you address a change, so I get notified, thanks

File: .github/workflows/verify.yaml (hunk 2 above)
Comment (mehmetihsansevinc) [reply .github/workflows/verify.yaml]: Inputs removed.

**Review by ndr-brt — APPROVED** [review]
LGTM

**Pull request merged**